### DISCUSSION TITLE
store target files in different subdirectories inside of s3?

### DISCUSSION THREAD

**Discussion by hershelm** [ropensci/targets#863] [Help] [answered]
Hey there, just getting started with using targets, and I was able to successfully set up the s3 backend.

Now, all files write into `_targets/objects/`

I was wondering whether it was possible to write directly to subdirectories inside of `_targets/objects`, e.g.

- `_targets/objects/data_raw/data.parquet`
- `_targets/objects/data_proc/data.parquet`
- `_targets/objects/plots/plot_1.png`

is this possible? Thanks!

**Comment by wlandau** [comment, marked as answer]
I wouldn't recommend modifying the structure `_targets/objects/` because the `targets` package controls all that internally, but you can set a `prefix` in `tar_resources_aws()` to store everything somewhere other than `_targets/objects`. For non-S3 targets, it is possible to define custom storage formats that create subdirectories of `_targets/objects/` (example: https://github.com/ropensci/targets/issues/861#issuecomment-1170498122), but I don't think it would work with S3 because that part of `targets` operates on one file at a time.

**Reply by hershelm** [reply]
thanks!
